lizrd/core/llm.py

```diff
@@ -325,11 +385,28 @@ def __init__(
                 layer.block_number = i_block
 
             _, current_device = self.get_current_device(i_block)
+            block = TransformerBlock(
+                dmodel,
+                layers_info,
+                gradient_checkpointing,
+                residual_fn,
+                wrap_attn_and_ff_in_fsdp=wrap_attn_and_ff_in_fsdp,
+                rank=rank,
+                param_precision=param_precision,
+                offload_params=offload_params,
+            )
+            if current_device != torch.device("cpu"):
+                block = block.to(current_device)
+            block = wrap_in_fsdp(
+                enabled=wrap_blocks_in_fsdp,
+                module=block,
+                rank=rank,
+                param_precision=param_precision,
+                offload_params=offload_params,
```

Comment:
this should not use `enabled`, but be under an if statement

Reply:
right